txsort.go

```diff
@@ -0,0 +1,90 @@
+// TxSort
+// Provides functions for sorting tx inputs and outputs according to BIP LI01
+
+package btcutil
+
+import (
+	"bytes"
+	"sort"
+
+	"github.com/btcsuite/btcd/wire"
+)
+
+// TxSort sorts the inputs and outputs of a tx based on BIP LI01
+// It does not modify the transaction given, but returns a new copy
+// which has been sorted and may have a different txid.
+func TxSort(tx *wire.MsgTx) *wire.MsgTx {
+	txCopy := tx.Copy()
+	sortInputs(txCopy.TxIn)
+	sortOutputs(txCopy.TxOut)
+	return txCopy
+}
+
+// TxIsSorted checks whether tx has inputs and outputs sorted according
+// to BIP LI01.
+func TxIsSorted(tx *wire.MsgTx) bool {
+	txCopy := tx.Copy()
```

Comment:
This implementation seem pretty expensive to me.

It would be much more efficient to loop the passed transaction and return false on the first case that violates the sort order.  This would result in a worst case run-time of `O(n)` versus this implementation which is `O(n^2)`.  In addition, it would avoid the copy, avoid the need to actually sort, and avoid the need to take the hash.

In fact, the main purpose for proposing this function was specifically to allow the caller to avoid having to do this.